src/solver.f90

```diff
@@ -93,37 +96,41 @@ function init(backend, time_integrator, host_allocator, &
     class(field_t), pointer :: u_init, v_init, w_init
 
     real(dp) :: x, y, z
-    integer :: nx, ny, nz, i, j, k
+    integer :: i, j, k
+    integer, dimension(3) :: dims
+    real(dp), dimension(3) :: xloc
 
     solver%backend => backend
+    solver%mesh = mesh
```

Comment:
Never experimented with copying user defined types in Fortran, are we sure that this is doing a 'deep copy'? Because mesh has some members which are also user defined types, are all of them copied into this new instance of the mesh class?

Also, would it be better if we point to the object we created in the main program? In the backend modules we actually point but here we copy instead, is there a reason for it?

Reply:
Oh yes, that's an oversight from my part, it should be a pointer:
```
solver%mesh => mesh
```
Otherwise indeed it will create a copy which we don't want. I will correct taht on monday